lib/wibox/layout/stack.lua

```diff
@@ -0,0 +1,188 @@
+-- @name insert
+-- @class function
+
+--- Remove one or more widgets from the layout
+-- The last parameter can be a boolean, forcing a recursive seach of the
+-- widget(s) to remove.
+-- @param widget ... Widgets that should be removed (must at least be one)
+-- @treturn boolean If the operation is successful
+-- @name remove_widgets
+-- @class function
+
+--- Add spacing between each layout widgets
+-- @param spacing Spacing between widgets.
+-- @name set_spacing
+-- @class function
+
+--- Layout a stack layout. Each widget get drawn on top of each other
+-- @param context The context in which we are drawn.
+-- @param width The available width.
+-- @param height The available height.
+function stack:layout(context, width, height)
+    local result = {}
+    local spacing = self._spacing
+
+    for k, v in pairs(self.widgets) do
+        table.insert(result, base.place_widget_at(v, spacing, spacing, width - 2*spacing, height - 2*spacing))
+        if self._top_only then break end
+    end
+
+    return result
+end
+
+--- Fit the stack layout into the given space
+-- @param context The context in which we are fit.
+-- @param orig_width The available width.
+-- @param orig_height The available height.
+function stack:fit(context, orig_width, orig_height)
+    local max_w, max_h = 0,0
+    local spacing = self._spacing
+
+    for k, v in pairs(self.widgets) do
+        local w, h = base.fit_widget(self, context, v, orig_width, orig_height)
+        max_w, max_h = math.max(max_w, w+2*spacing), math.max(max_h, h+2*spacing)
+    end
+
+    return math.min(max_w, orig_width), math.min(max_h, orig_height)
+end
+
+--- Get if only the first stack widget is drawn
+-- @return If the only the first stack widget is drawn
+function stack:get_display_top_only()
+    return self._top_only
+end
+
+--- Only draw the first widget of the stack, ignore others
+-- @tparam boolean top_only Only draw the top stack widget
+function stack:set_display_top_only(top_only)
+    self._top_only = top_only
+end
+
+--- Raise a widget at `index` to the top of the stack
+-- @tparam number index the widget index to raise
+function stack:raise(index)
+    if (not index) or self.widgets[index] then return end
+
+    local w = self.widgets[index]
+    table.remove(self.widgets, index)
+    table.insert(self.widgets, w)
+
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Raise the first instance of `widget`
+-- @param widget The widget to raise
+-- @tparam[opt=false] boolean recursive Also look deeper in the hierarchy to
+--   find the widget
+function stack:raise_widget(widget, recursive)
+    local idx, layout = self:index(widget, recursive)
+
+    if not idx or not layout then return end
+
+    -- Bubble up in the stack until the right index is found
+    while layout and layout ~= self do
+        idx, layout = self:index(layout, recursive)
+    end
+
+    if layout == self and idx ~= 1 then
+        self:raise(idx)
+    end
+end
+
+local function new(dir, widget1, ...)
```

Comment:
Luacheck says:

    lib/wibox/layout/stack.lua:175:20: unused argument dir
    lib/wibox/layout/stack.lua:175:25: unused argument widget1

Could you please add APIdocs to something explaining the arguments and fixing the above? (From these warnings it looks like this just shouldn't get any arguments at all...)